SwiftCompilerSources/Sources/Optimizer/PassManager/Context.swift

```diff
@@ -437,8 +464,19 @@ extension Builder {
               context.notifyInstructionChanged, context._bridged.asNotificationHandler())
   }
 
-  /// Creates a builder which inserts _before_ `insPnt`, using the location of `insPnt`.
+  /// Creates a builder which inserts _before_ `insPnt`, using the location of `insPnt`,
+  /// skipping meta instructions.
   init(before insPnt: Instruction, _ context: some MutatingContext) {
+    context.verifyIsTransforming(function: insPnt.parentFunction)
+    self.init(insertAt: .before(insPnt),
+              location: insPnt.locationSkippingMetaInstructions,
+              context.notifyInstructionChanged, context._bridged.asNotificationHandler())
+  }
+
+  /// Creates a builder which inserts _before_ `insPnt`, using the exact location of `insPnt`,
```

Comment:
We really need to clearly document when to use `insert(before:)` and `insert(replacing:)`.
You already gave a good explanation in the other PR:

> When we know that we have a debug_value, and want to replace it with a new one -> we want to take the debug_value's scope
> When we are moving code somewhere, on an unknown instruction, we want to ignore any alloc_stack or debug_value that might be there [...]

Can you add a comment like this here?

Also, if it's really just about `debug_value` (or meta instructions?) does it make sense to make this explicit in the signature?
```
init(replacing insPnt: DebugValueInst, _ context: some MutatingContext)
```
or
```
init(replacing insPnt: MetaInstruction, _ context: some MutatingContext)
```